web/src/pages/billing/BillingCostByCategory.tsx

```diff
@@ -93,24 +176,183 @@ const BillingCostByCategory: React.FunctionComponent = () => {
     const onGroupBySelect = (event: any, recs: any) => {
         setGroupBy(recs.value)
         setSelectedGroup(undefined)
-        updateNav(recs.value, undefined, selectedCostCategory, selectedPeriod, start, end)
+        // Clear filters when switching groupBy
+        if (recs.value !== BillingColumn.GcpProject) {
+            setSelectedProjects([])
+        }
+        if (recs.value !== BillingColumn.Topic) {
+            setSelectedTopics([])
+        }
+        updateNav(
+            recs.value,
+            undefined,
+            selectedCostCategory,
+            selectedPeriod,
+            start,
+            end,
+            recs.value === BillingColumn.GcpProject ? selectedProjects : [],
+            recs.value === BillingColumn.Topic ? selectedTopics : []
+        )
+        // Show loading immediately to prevent flash of old data
+        if (Boolean(start) && Boolean(end)) {
+            setIsLoading(true)
+            const query = buildFilteredQueryModel()
```

Comment:
These changes to use `buildFilteredQueryModel()` are relying on the updated state being available immediately, which won't be the case. You'll notice that when testing this in the UI whenever you select something it will use the previous value. You'll need to pass the updated value along to the call to buildFilteredQueryModel.
See this bit of the docs for more info: https://react.dev/reference/react/useState#ive-updated-the-state-but-logging-gives-me-the-old-value